owncloudComLibrary/src/main/java/com/owncloud/android/lib/resources/shares/ShareToRemoteOperationResultParser.kt

```diff
@@ -0,0 +1,142 @@
+/* ownCloud Android Library is available under MIT license
+ *   @author David A. Velasco
+ *   @author David González Verdugo
+ *   @author Christian Schabesberger
+ *   Copyright (C) 2019 ownCloud GmbH.
+ *
+ *   Permission is hereby granted, free of charge, to any person obtaining a copy
+ *   of this software and associated documentation files (the "Software"), to deal
+ *   in the Software without restriction, including without limitation the rights
+ *   to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ *   copies of the Software, and to permit persons to whom the Software is
+ *   furnished to do so, subject to the following conditions:
+ *
+ *   The above copyright notice and this permission notice shall be included in
+ *   all copies or substantial portions of the Software.
+ *
+ *   THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+ *   EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+ *   MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+ *   NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS
+ *   BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN
+ *   ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+ *   CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ *   THE SOFTWARE.
+ *
+ */
+
+package com.owncloud.android.lib.resources.shares
+
+import android.net.Uri
+
+import com.owncloud.android.lib.common.operations.RemoteOperationResult
+import com.owncloud.android.lib.common.utils.Log_OC
+import com.owncloud.android.lib.resources.status.OwnCloudVersion
+import org.xmlpull.v1.XmlPullParserException
+
+import java.io.ByteArrayInputStream
+import java.io.IOException
+import java.io.InputStream
+import java.util.ArrayList
+
+class ShareToRemoteOperationResultParser(shareXmlParser: ShareXMLParser) {
+
+    private var mShareXmlParser: ShareXMLParser? = null
+    private var mOneOrMoreSharesRequired = false
+    private var mOwnCloudVersion: OwnCloudVersion? = null
+    private var mServerBaseUri: Uri? = null
+
+    init {
+        mShareXmlParser = shareXmlParser
```

Comment:
why can mShareXmlParser be nullable and shareXmlParser cannot? I think mShareXmlParser cannot be nullable either.

Reply:
BTW, if shareXmlParser can be nullable and mShareXmlParser cannot. why don't you refactor init block to this:

mShareXmlParser = shareXmlParser ?: ShareXMLParser()